### PR TITLE
FIX: Fix incorrect scale of base_score being used for XGB regression objectives

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR fixes an issue with the daal4py model builders using the intercepts from XGBoost models incorrectly when it comes to regression models that have a link function, such as Poisson or Gamma.

Compared to the other GBT libraries supported by model builders, XGBoost has a 'base_score' field which is in the scale of the response variable (i.e. inverse link function is applied to it), but the model builders assume that the intercept is on the link scale, so predictions will be wrong when there is a non-identity link function.

Along the way, it also fixes an issue of the intercept not being applied to objective `binary:logitraw` due to this same issue, which was previously being approach through workarounds that remove the intercept.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

Not applicable.

### CODE REVIEW

**Review by Alexsandruss — APPROVED** [review]

**Review by razdoburdin — APPROVED** [review]
Great!
